hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl

```diff
@@ -32,42 +32,51 @@ proc generate_mmi {filename brams designtask_count} {
         if {$slice_begin eq {} || $slice_end eq {}} {
             send_msg "${designtask_count}-2" ERROR "Extraction of ${filename} information failed."
         }
+
         # The scrambled Boot ROM is actually 39 bits wide but the updatemem tool segfaults
         # for slice sizes not divisible by 8.
-        if {[expr {($slice_end - $slice_begin + 1) < 8}]} {
+        if {$filename eq "rom.mmi" && [expr {($slice_end - $slice_begin + 1) < 8}]} {
```

Comment:
It kind of looks like we're lying to `updatemem` and saying that the bitlanes cover 40 bits when they actually cover 39. I wonder if this was because you can't write 39 bits with 10 hex digits. Maybe this special case for ROM would not be necessary if we wrote four 39-bit words at a time (39 hex digits)?